In [1]:
"""
    @date 12/02/19
    @author Florian "lefko" Becker
    @summary part I: exploratory data analysis
"""

'\n    @date 12/02/19\n    @author Florian "lefko" Becker\n    @summary part I: exploratory data analysis\n'

In [8]:
# necessary imports
import pandas as pd
import os

In [9]:
# current working directory
os.getcwd()

'/home/lefko/git'

In [22]:
# load the bchi dataset; this is the complete dataset, ~ 16 MB of size
bchi_df = pd.read_csv('./ut-health-project/data/BCHI-dataset_2019-03-04.csv')
print(len(bchi_df))

34492


In [11]:
bchi_df.head(10) # show first 10 rows

Indicator Category  \
0  Behavioral Health/Substance Abuse   
1  Behavioral Health/Substance Abuse   
2  Behavioral Health/Substance Abuse   
3  Behavioral Health/Substance Abuse   
4  Behavioral Health/Substance Abuse   
5  Behavioral Health/Substance Abuse   
6  Behavioral Health/Substance Abuse   
7  Behavioral Health/Substance Abuse   
8  Behavioral Health/Substance Abuse   
9  Behavioral Health/Substance Abuse   

                                           Indicator  Year   Sex  \
0  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
1  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
2  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
3  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
4  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
5  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
6  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
7  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
8  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
9  Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   

  Race/Ethnicity  Value                            Place  \
0            All    1.7                   Washington, DC   
1            All    2.2  Fort Worth (Tarrant County), TX   
2            All    2.3     Oakland (Alameda County), CA   
3            All    3.0                  San Antonio, TX   
4            All    4.4           U.S. Total, U.S. Total   
5            All    5.2             San Diego County, CA   
6            All    5.4                  Kansas City, MO   
7            All    6.3                       Denver, CO   
8            All   11.3     Las Vegas (Clark County), NV   
9            All   11.8                     Columbus, OH   

                          BCHC Requested Methodology  \
0  Age-Adjusted rate of opioid-related mortality ...   
1  Age-adjusted rate of opioid-related mortality ...   
2  Age-adjusted rate of opioid-related mortality ...   
3  Age-adjusted rate of opioid-related mortality ...   
4  Age-Adjusted rate of opioid-related mortality ...   
5  Age-adjusted rate of opioid-related mortality ...   
6  Age-adjusted rate of opioid-related mortality ...   
7  Age-adjusted rate of opioid-related mortality ...   
8  Age-adjusted rate of opioid-related mortality ...   
9  Age-Adjusted rate of opioid-related mortality ...   

                                              Source  \
0  D.C. Department of Health, Center for Policy, ...   
1              National Center for Health Statistics   
2                                         CDC Wonder   
3                                         CDC Wonder   
4                                         CDC WONDER   
5  Centers for Disease Control and Prevention, Na...   
6                                                NaN   
7  Mortality data from the Colorado Department of...   
8         Nevada Vital Records - Clark County Deaths   
9  Mortality data from Ohio Department of Health,...   

                                             Methods  \
0                                                NaN   
1                                                NaN   
2  Age-adjusted rate of opioid-related mortality ...   
3                                                NaN   
4                                                NaN   
5  All data was pulled using the CDC WONDER morta...   
6                                                NaN   
7                                                NaN   
8                                                NaN   
9                                                NaN   

                                               Notes  \
0  This indicator is not exclusive of other drugs...   
1  This indicator is not exclusive of other drugs...   
2  Data is for Alameda County. This indicator is ...   
3  This indicator is not exclusive of other drugs...   
4  This indicator is not exclusive o

In [13]:
# list unique values of 'Year' and 'Place'
years = bchi_df.Year.unique()
places = bchi_df.Place.unique()

print(years, places)

[2010 2011 2012 2013 2014 2015 2016 2018 2017] ['Washington, DC' 'Fort Worth (Tarrant County), TX'
 'Oakland (Alameda County), CA' 'San Antonio, TX' 'U.S. Total, U.S. Total'
 'San Diego County, CA' 'Kansas City, MO' 'Denver, CO'
 'Las Vegas (Clark County), NV' 'Columbus, OH' 'Phoenix, AZ' 'Boston, MA'
 'Houston, TX' 'Minneapolis, MN' 'Los Angeles, CA'
 'Miami (Miami-Dade County), FL' 'Portland (Multnomah County), OR'
 'Charlotte, NC' 'Baltimore, MD' 'Philadelphia, PA' 'Seattle, WA'
 'Detroit, MI' 'New York City, NY' 'Long Beach, CA' 'Chicago, Il'
 'Indianapolis (Marion County), IN' 'San Jose, CA' 'San Francisco, CA'
 'Cleveland, OH' 'Dallas, TX' 'Austin, TX']


The problem is, that not for every year there is data available for every city, so now let's find those combinations of city/year, where both are found

In [42]:
cities_grouped_per_rows = bchi_df.groupby('Place').Year.count().reset_index(name = 'count').sort_values(['count'], ascending = False)
print(cities_grouped_per_rows[0:5])
top5_cities = cities_grouped_per_rows[0:5].Place

Place  count
25          San Diego County, CA   1913
8                     Denver, CO   1805
14  Las Vegas (Clark County), NV   1780
24               San Antonio, TX   1618
2                     Boston, MA   1565


So, we should focus only on the top 5 of the cities, since then we can assume, that there should be data available for all the years

In [52]:
bchi_df_filtered = bchi_df[(bchi_df.Year.isin(years) & bchi_df.Place.isin(top5_cities)) == True]
len(bchi_df_filtered)
print(bchi_df_filtered)

Indicator Category  \
3      Behavioral Health/Substance Abuse   
5      Behavioral Health/Substance Abuse   
7      Behavioral Health/Substance Abuse   
8      Behavioral Health/Substance Abuse   
12     Behavioral Health/Substance Abuse   
...                                  ...   
34467        Social and Economic Factors   
34473        Social and Economic Factors   
34477        Social and Economic Factors   
34481        Social and Economic Factors   
34485        Social and Economic Factors   

                                               Indicator  Year   Sex  \
3      Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
5      Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
7      Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
8      Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
12     Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
...                                                  ...   ..

Now only 8681 rows remain. In the next step we'll check the columns to see if some are redundant or unecessary

In [53]:
bchi_df_filtered.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8681 entries, 3 to 34485
Data columns (total 15 columns):
Indicator Category             8681 non-null object
Indicator                      8681 non-null object
Year                           8681 non-null int64
Sex                            8681 non-null object
Race/Ethnicity                 8681 non-null object
Value                          7492 non-null float64
Place                          8681 non-null object
BCHC Requested Methodology     8681 non-null object
Source                         7579 non-null object
Methods                        2678 non-null object
Notes                          4378 non-null object
90% Confidence Level - Low     638 non-null float64
90% Confidence Level - High    638 non-null float64
95% Confidence Level - Low     2744 non-null float64
95% Confidence Level - High    2744 non-null float64
dtypes: float64(5), int64(1), object(9)
memory usage: 1.1+ MB


Let's drop the following columns: 'BCHC Requested Methodology, Source, Methods, Notes' and these 'Confidence Level' columns

In [54]:
bchi_df_filtered = bchi_df_filtered.drop(columns = ['BCHC Requested Methodology', 'Source', 'Methods', 'Notes', '90% Confidence Level - Low', '90% Confidence Level - High', '95% Confidence Level - Low', '95% Confidence Level - High'], axis = 1)

In [55]:
bchi_df_filtered.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8681 entries, 3 to 34485
Data columns (total 7 columns):
Indicator Category    8681 non-null object
Indicator             8681 non-null object
Year                  8681 non-null int64
Sex                   8681 non-null object
Race/Ethnicity        8681 non-null object
Value                 7492 non-null float64
Place                 8681 non-null object
dtypes: float64(1), int64(1), object(5)
memory usage: 862.6+ KB


In [56]:
# check for NaNs
bchi_df_filtered.isna().sum()

Indicator Category       0
Indicator                0
Year                     0
Sex                      0
Race/Ethnicity           0
Value                 1189
Place                    0
dtype: int64

1189 missing values in the 'Value' column. Since there are of no use for us we'll remove these rows.

In [64]:
bchi_df_filtered.dropna() # 7492 rows remain

Indicator Category  \
3      Behavioral Health/Substance Abuse   
5      Behavioral Health/Substance Abuse   
7      Behavioral Health/Substance Abuse   
8      Behavioral Health/Substance Abuse   
12     Behavioral Health/Substance Abuse   
...                                  ...   
34467        Social and Economic Factors   
34473        Social and Economic Factors   
34477        Social and Economic Factors   
34481        Social and Economic Factors   
34485        Social and Economic Factors   

                                               Indicator  Year   Sex  \
3      Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
5      Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
7      Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
8      Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
12     Opioid-Related Unintentional Drug Overdose Mor...  2010  Both   
...                                                  ...   ...   ...   
34467                                 Percent Unemployed  2014  Male   
34473                                 Percent Unemployed  2014  Male   
34477                                 Percent Unemployed  2014  Male   
34481                                 Percent Unemployed  2014  Male   
34485                                 Percent Unemployed  2014  Male   

      Race/Ethnicity  Value                         Place  
3                All    3.0               San Antonio, TX  
5                All    5.2          San Diego County, CA  
7                All    6.3                    Denver, CO  
8                All   11.3  Las Vegas (Clark County), NV  
12          Asian/PI    0.0                    Denver, CO  
...              ...    ...                           ...  
34467            All    4.4                    Denver, CO  
34473            All    6.3               San Antonio, TX  
34477            All    7.1          San Diego County, CA  
34481            All    7.9                    Boston, MA  
34485            All    8.7  Las Vegas (Clark County), NV  

[7492 rows x 7 columns]

In [65]:
bchi_df_filtered.Indicator.unique() # pick useful indicators; they should be comparable with a 

array(['Opioid-Related Unintentional Drug Overdose Mortality Rate (Age-Adjusted; Per 100,000 people)',
       'Percent of Adults Who Binge Drank',
       'Percent of High School Students Who Binge Drank',
       'All Types of Cancer Mortality Rate (Age-Adjusted; Per 100,000 people)',
       'Female Breast Cancer Mortality Rate (Age-Adjusted; Per 100,000 people)',
       'Lung Cancer Mortality Rate (Age-Adjusted; Per 100,000 people)',
       'Asthma Emergency Department Visit Rate (Age-Adjusted; Per 10,000)',
       'Diabetes Mortality Rate (Age-Adjusted; Per 100,000 people)',
       'Heart Disease Mortality Rate (Age-Adjusted; Per 100,000 people)',
       'Percent of Adults Who Are Obese',
       'Percent of Adults Who Currently Smoke',
       'Percent of Adults Who Meet CDC-Recommended Physical Activity Levels',
       'Percent of High School Students Who Are Obese',
       'Percent of High School Students Who Currently Smoke',
       'Percent of High School Students Who Meet CDC-Reco